language-support/ts/daml-ledger/index.ts

```diff
@@ -517,12 +536,30 @@ class Ledger {
       } else {
         console.error('Ledger.streamQuery unknown message', json);
       }
-    });
+    };
+    const onClose = ({ code, reason }: {
+        wasClean: boolean; code: number;
+        reason: string; target: WebSocket;
+    }): void => {
+      emitter.emit('close', {code, reason});
+      const now = new Date().getTime();
+      // we try to reconnect if we could connect previously and we were live for at least
+      // 'reconnectThreshold'.
+      if (lastOffset && isLiveSince && now - isLiveSince >= this.reconnectThreshold) {
+        console.log('reconnecting WebSocket');
```

Comment:
We shouldn't have this in prod code. If you think it's useful, please use `console.debug` and add a more descriptive message that tells you which websocket is reconnecting.
```suggestion
```